[AutoDiff] Crash due to a use after consuming `differentiable_function`

### Description

Crash appears when optimization is enabled (`swift-frontend -emit-sil -O`) during CopyPropagation due to a `debug_value` use of a value previously consumed by `differentiable_function`.

The incorrect SIL with this use (corresponding to the Swift reproducer):

<details>

```
// foo()
// Isolation: unspecified
sil [ossa] @$s3src3fooyyF : $@convention(thin) () -> () {
[global: read,write,copy,destroy,allocate,deinit_barrier]
bb0:
  // function_ref closure #1 in foo()
  %0 = function_ref @$s3src3fooyyFS2dcfU_ : $@convention(thin) (Double) -> Double // user: %3
  // function_ref forward-mode derivative of closure #1 in foo()
  %1 = function_ref @$s3src3fooyyFS2dcfU_TJfSpSr : $@convention(thin) (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double) // user: %4
  // function_ref reverse-mode derivative of closure #1 in foo()
  %2 = function_ref @$s3src3fooyyFS2dcfU_TJrSpSr : $@convention(thin) (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double) // user: %5
  %3 = thin_to_thick_function %0 to $@noescape @callee_guaranteed (Double) -> Double // users: %35, %8, %6; ownership: none
  %4 = thin_to_thick_function %1 to $@noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double) // users: %14, %6; ownership: none
  %5 = thin_to_thick_function %2 to $@noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double) // users: %20, %6; ownership: none
  %6 = differentiable_function [parameters 0] [results 0] %3 with_derivative {%4, %5} // user: %36
  // function_ref thunk for @callee_guaranteed (@unowned Double) -> (@unowned Double)
  %7 = function_ref @$sS2dIgyd_S2dIegnr_TR : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> Double) -> @out Double // users: %35, %8
  %8 = partial_apply [callee_guaranteed] %7(%3) : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> Double) -> @out Double // user: %9
  %9 = convert_function %8 to $@callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double> // users: %11, %10
  %10 = copy_value %9                             // users: %41, %12
  destroy_value %9                                // id: %11
  %12 = convert_escape_to_noescape %10 to $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double> // users: %28, %25
  // function_ref thunk for @callee_guaranteed (@unowned Double) -> (@unowned Double, @owned @escaping @callee_guaranteed (@unowned Double) -> (@unowned Double))
  %13 = function_ref @$sS4dIegyd_Igydo_S2dxq_Ri_zRi0_zRi__Ri0__r0_lyS2dIsegnr_Iegnro_TR : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)) -> (@out Double, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>) // user: %14
  %14 = partial_apply [callee_guaranteed] %13(%4) : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)) -> (@out Double, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>) // user: %15
  %15 = convert_function %14 to $@callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double> // users: %17, %16
  %16 = copy_value %15                            // users: %42, %18
  destroy_value %15                               // id: %17
  %18 = convert_escape_to_noescape %16 to $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double> // user: %25
  // function_ref thunk for @callee_guaranteed (@unowned Double) -> (@unowned Double, @owned @escaping @callee_guaranteed (@unowned Double) -> (@unowned Double))
  %19 = function_ref @$sS4dIegyd_Igydo_S2dxq_Ri_zRi0_zRi__Ri0__r0_lyS2dIsegnr_Iegnro_TR : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)) -> (@out Double, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>) // user: %20
  %20 = partial_apply [callee_guaranteed] %19(%5) : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)) -> (@out Double, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>) // user: %21
  %21 = convert_function %20 to $@callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double> // users: %23, %22
  %22 = copy_value %21                            // users: %43, %24
  destroy_value %21                               // id: %23
  %24 = convert_escape_to_noescape %22 to $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double> // user: %25
  %25 = differentiable_function [parameters 0] [results 0] %12 with_derivative {%18, %24} // users: %37, %29
  %26 = float_literal $Builtin.FPIEEE64, 0x0 // 0 // user: %27
  %27 = struct $Double (%26)                      // users: %30, %33
  debug_value %12, let, name "f", argno 1         // id: %28
  debug_value %25, let, name "f", argno 1         // id: %29
  debug_value %27, let, name "x", argno 2         // id: %30
  %31 = alloc_stack $Double                       // users: %35, %39
  %32 = alloc_stack $Double                       // users: %35, %38, %34, %33
  store %27 to [trivial] %32                      // id: %33
  debug_value %32, let, name "x", argno 2, expr op_deref // id: %34
  %35 = apply %7(%31, %32, %3) : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> Double) -> @out Double
  destroy_value %6                                // id: %36
  destroy_value %25                               // id: %37
  dealloc_stack %32                               // id: %38
  dealloc_stack %31                               // id: %39
  %40 = tuple ()                                  // user: %44
  destroy_value %10                               // id: %41
  destroy_value %16                               // id: %42
  destroy_value %22                               // id: %43
  return %40                                      // id: %44
} // end sil function '$s3src3fooyyF'
```

</details>

### Reproduction

**Note:** it's important to have generics - w/o them, the crash disappears

```swift
// swift-frontend -emit-sil -O
import _Differentiation

func compute<T>(_ f: (T) -> T, _ x: T) -> T {
  return f(x)
}

func checkDerivative<T: Differentiable>(
  _ f: @differentiable(reverse) (T) -> T,
  _ x: T)
{
  compute(f, x)
}

public func foo() {
  checkDerivative({ $0 }, 0.0)
}
```


### Stack dump

```text
egin Error in Function: '$s3src3fooyyF'
Found outside of lifetime use?!
Value:   %12 = convert_escape_to_noescape %10 : $@callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double> to $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double> // users: %28, %25
Consuming User:   %25 = differentiable_function [parameters 0] [results 0] %12 : $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double> with_derivative {%18 : $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double>, %24 : $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double>} // users: %37, %29
Non Consuming User:   debug_value %12 : $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>, let, name "f", argno 1 // id: %28
Block: bb0

End Error in Function: '$s3src3fooyyF'
Found ownership error?!
/path/to/src.swift:15:3: warning: result of call to 'compute' is unused [#NoUsage]
13 |   _ x: T)
14 | {
15 |   compute(f, x)
   |   `- warning: result of call to 'compute' is unused [#NoUsage]
16 | }
17 |

<unknown>:0: error: fatal error encountered during compilation; please submit a bug report (https://swift.org/contributing/#reporting-bugs)
<unknown>:0: note: triggering standard assertion failure routine
Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the crash backtrace.
Stack dump:
0.      Program arguments: ../build/Ninja-RelWithDebInfoAssert/swift-linux-x86_64/bin/swift-frontend -emit-sil -O /path/to/src.swift
1.      Swift version 6.4-dev (LLVM 155bd39530126ad, Swift 2e033e9bb76b0d4)
2.      Compiling with effective version 5.10
3.      While evaluating request ExecuteSILPipelineRequest(Run pipelines { PrepareOptimizationPasses, EarlyModulePasses, HighLevel,Function+EarlyLoopOpt, HighLevel,Module+StackPromote, MidLevel,Function, ClosureSpecialize, LowLevel,Function, LateLoopOpt, SIL Debug Info Generator } on SIL for src)
4.      While running pass #471 SILFunctionTransform "CopyPropagation" on SILFunction "@$s3src3fooyyF".
 for 'foo()' (at /path/to/src.swift:18:8)
 #0 0x000001fcc3e26628 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) /path/to/llvm-project/llvm/lib/Support/Unix/Signals.inc:834:13
 #1 0x000001fcc3e24153 llvm::sys::RunSignalHandlers() /path/to/llvm-project/llvm/lib/Support/Signals.cpp:105:18
 #2 0x000001fcc3e27341 SignalHandler(int, siginfo_t*, void*) /path/to/llvm-project/llvm/lib/Support/Unix/Signals.inc:426:38
 #3 0x0000644c9363df80 (/usr/lib/libc.so.6+0x3df80)
 #4 0x0000644c9369630c __pthread_kill_implementation /usr/src/debug/glibc/glibc/nptl/pthread_kill.c:44:76
 #5 0x0000644c9363de48 raise /usr/src/debug/glibc/glibc/signal/../sysdeps/posix/raise.c:27:6
 #6 0x0000644c936256c9 abort /usr/src/debug/glibc/glibc/stdlib/abort.c:85:3
 #7 0x000001fcbc7e887b llvm::ThreadSafeRefCountedBase<llvm::vfs::FileSystem>::Release() const /path/to/llvm-project/llvm/include/llvm/ADT/IntrusiveRefCntPtr.h:135:5
 #8 0x000001fcbc7e887b llvm::IntrusiveRefCntPtrInfo<llvm::vfs::FileSystem>::release(llvm::vfs::FileSystem*) /path/to/llvm-project/llvm/include/llvm/ADT/IntrusiveRefCntPtr.h:164:38
 #9 0x000001fcbc7e887b llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>::release() /path/to/llvm-project/llvm/include/llvm/ADT/IntrusiveRefCntPtr.h:233:7
#10 0x000001fcbc7e887b llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>::~IntrusiveRefCntPtr() /path/to/llvm-project/llvm/include/llvm/ADT/IntrusiveRefCntPtr.h:196:27
#11 0x000001fcbc7e887b swift::DiagnosticHelper::Implementation::diagnoseFatalError(char const*, bool) /path/to/swift/lib/Frontend/DiagnosticHelper.cpp:484:17
#12 0x000001fcc3d7c1b6 std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>::_M_data() const /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/15.2.1/../../../../include/c++/15.2.1/bits/basic_string.h:239:28
#13 0x000001fcc3d7c1b6 std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>::_M_is_local() const /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/15.2.1/../../../../include/c++/15.2.1/bits/basic_string.h:280:6
#14 0x000001fcc3d7c1b6 std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>::_M_dispose() /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/15.2.1/../../../../include/c++/15.2.1/bits/basic_string.h:298:7
#15 0x000001fcc3d7c1b6 std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>::~basic_string() /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/15.2.1/../../../../include/c++/15.2.1/bits/basic_string.h:896:9
#16 0x000001fcc3d7c1b6 llvm::report_fatal_error(llvm::Twine const&, bool) /path/to/llvm-project/llvm/lib/Support/ErrorHandling.cpp:117:5
#17 0x000001fcc3d7c0a6 (../build/Ninja-RelWithDebInfoAssert/swift-linux-x86_64/bin/swift-frontend+0x838b0a6)
#18 0x000001fcbd2a289b swift::LinearLifetimeChecker::ErrorBehaviorKind::shouldReturnFalse() const /path/to/swift/lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h:61:5
#19 0x000001fcbd2a289b swift::LinearLifetimeChecker::ErrorBuilder::handleError(llvm::function_ref<void ()>&&, bool) const /path/to/swift/lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h:206:18
#20 0x000001fcbd31b8c5 swift::LinearLifetimeChecker::ErrorBuilder::handleUseOutsideOfLifetime(llvm::function_ref<void ()>&&) /path/to/swift/lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h:0:5
#21 0x000001fcbd31b8c5 (anonymous namespace)::State::checkForSameBlockUseAfterFree(swift::Operand*, swift::SILBasicBlock*) /path/to/swift/lib/SIL/Verifier/LinearLifetimeChecker.cpp:400:18
#22 0x000001fcbd31b8c5 (anonymous namespace)::State::initializeAllConsumingUses(llvm::ArrayRef<swift::Operand*>, llvm::SmallVectorImpl<std::pair<swift::Operand*, swift::SILBasicBlock*>>&) /path/to/swift/lib/SIL/Verifier/LinearLifetimeChecker.cpp:332:5
#23 0x000001fcbd31b8c5 swift::LinearLifetimeChecker::checkValueImpl(swift::SILValue, llvm::ArrayRef<swift::Operand*>, llvm::ArrayRef<swift::Operand*>, swift::LinearLifetimeChecker::ErrorBuilder&, std::optional<llvm::function_ref<void (swift::SILBasicBlock*)>>, std::optional<llvm::function_ref<void (swift::Operand*)>>) /path/to/swift/lib/SIL/Verifier/LinearLifetimeChecker.cpp:695:9
#24 0x000001fcbd31d23d swift::LinearLifetimeChecker::checkValue(swift::SILValue, llvm::ArrayRef<swift::Operand*>, llvm::ArrayRef<swift::Operand*>, swift::LinearLifetimeChecker::ErrorBuilder&) /path/to/swift/lib/SIL/Verifier/LinearLifetimeChecker.cpp:773:10
#25 0x000001fcbd29f588 swift::LinearLifetimeChecker::Error::getFoundError() const /path/to/swift/lib/SIL/Verifier/LinearLifetimeCheckerPrivate.h:78:30
#26 0x000001fcbd29f588 swift::SILValueOwnershipChecker::check() /path/to/swift/lib/SIL/Verifier/SILOwnershipVerifier.cpp:193:34
#27 0x000001fcbd2a2034 llvm::SmallVectorTemplateCommon<swift::Operand*, void>::begin() /path/to/llvm-project/llvm/include/llvm/ADT/SmallVector.h:268:45
#28 0x000001fcbd2a2034 llvm::SmallVectorTemplateCommon<swift::Operand*, void>::end() /path/to/llvm-project/llvm/include/llvm/ADT/SmallVector.h:270:27
#29 0x000001fcbd2a2034 llvm::SmallVector<swift::Operand*, 16u>::~SmallVector() /path/to/llvm-project/llvm/include/llvm/ADT/SmallVector.h:1203:46
#30 0x000001fcbd2a2034 swift::SILValueOwnershipChecker::~SILValueOwnershipChecker() /path/to/swift/lib/SIL/Verifier/SILOwnershipVerifier.cpp:127:39
#31 0x000001fcbd2a2034 verifySILValueHelper(swift::SILFunction const*, swift::SILValue, swift::LinearLifetimeChecker::ErrorBuilder&, swift::DeadEndBlocks*, swift::GuaranteedPhiVerifier&) /path/to/swift/lib/SIL/Verifier/SILOwnershipVerifier.cpp:962:3
#32 0x000001fcbd2a24fe swift::SILInstructionResultArray::iterator::operator++() /path/to/swift/include/swift/SIL/SILInstruction.h:339:5
#33 0x000001fcbd2a24fe swift::SILFunction::verifyOwnership(swift::DeadEndBlocks*) const /path/to/swift/lib/SIL/Verifier/SILOwnershipVerifier.cpp:1067:24
#34 0x000001fcbd41e317 std::_Function_base::~_Function_base() /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/std_function.h:248:11
#35 0x000001fcbd41e317 swift::InstModCallbacks::~InstModCallbacks() /path/to/swift/include/swift/SILOptimizer/Utils/InstModCallbacks.h:139:31
#36 0x000001fcbd41e317 swift::InstructionDeleter::~InstructionDeleter() /path/to/swift/include/swift/SILOptimizer/Utils/InstructionDeleter.h:112:7
#37 0x000001fcbd41e317 (anonymous namespace)::CopyPropagation::run() /path/to/swift/lib/SILOptimizer/Transforms/CopyPropagation.cpp:690:1
#38 0x000001fcbd360a9d swift::SILPassManager::runPassOnFunction(unsigned int, swift::SILFunction*) /path/to/swift/lib/SILOptimizer/PassManager/PassManager.cpp:729:25
#39 0x000001fcbd3622fb std::vector<swift::SILPassManager::WorklistEntry, std::allocator<swift::SILPassManager::WorklistEntry>>::size() const /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/stl_vector.h:1109:60
#40 0x000001fcbd3622fb std::vector<swift::SILPassManager::WorklistEntry, std::allocator<swift::SILPassManager::WorklistEntry>>::operator[](unsigned long) /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/stl_vector.h:1253:2
#41 0x000001fcbd3622fb swift::SILPassManager::runFunctionPasses(unsigned int, unsigned int) /path/to/swift/lib/SILOptimizer/PassManager/PassManager.cpp:871:28
#42 0x000001fcbd365ea3 swift::SILPassManager::execute() /path/to/swift/lib/SILOptimizer/PassManager/PassManager.cpp:1061:1
#43 0x000001fcbd35e418 __gnu_cxx::__normal_iterator<swift::SILPassPipeline const*, std::vector<swift::SILPassPipeline, std::allocator<swift::SILPassPipeline>>>::operator++() /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/stl_iterator.h:1103:2
#44 0x000001fcbd35e418 swift::SILPassManager::executePassPipelinePlan(swift::SILPassPipelinePlan const&) /path/to/swift/lib/SILOptimizer/PassManager/PassManager.cpp:1003:40
#45 0x000001fcbd35e3cb swift::ExecuteSILPipelineRequest::evaluate(swift::Evaluator&, swift::SILPipelineExecutionDescriptor) const /path/to/swift/lib/SILOptimizer/PassManager/PassManager.cpp:395:1
#46 0x000001fcbd382c8a std::tuple<> swift::SimpleRequest<swift::ExecuteSILPipelineRequest, std::tuple<> (swift::SILPipelineExecutionDescriptor), (swift::RequestFlags)1>::callDerived<0ul>(swift::Evaluator&, std::integer_sequence<unsigned long, 0ul>) const /path/to/swift/include/swift/AST/SimpleRequest.h:287:24
#47 0x000001fcbd382c8a swift::SimpleRequest<swift::ExecuteSILPipelineRequest, std::tuple<> (swift::SILPipelineExecutionDescriptor), (swift::RequestFlags)1>::evaluateRequest(swift::ExecuteSILPipelineRequest const&, swift::Evaluator&) /path/to/swift/include/swift/AST/SimpleRequest.h:311:20
#48 0x000001fcbd36de83 llvm::DenseMapBase<llvm::DenseMap<swift::ActiveRequest, llvm::detail::DenseSetEmpty, llvm::DenseMapInfo<swift::ActiveRequest, void>, llvm::detail::DenseSetPair<swift::ActiveRequest>>, swift::ActiveRequest, llvm::detail::DenseSetEmpty, llvm::DenseMapInfo<swift::ActiveRequest, void>, llvm::detail::DenseSetPair<swift::ActiveRequest>>::empty() const /path/to/llvm-project/llvm/include/llvm/ADT/DenseMap.h:118:61
#49 0x000001fcbd36de83 llvm::detail::DenseSetImpl<swift::ActiveRequest, llvm::DenseMap<swift::ActiveRequest, llvm::detail::DenseSetEmpty, llvm::DenseMapInfo<swift::ActiveRequest, void>, llvm::detail::DenseSetPair<swift::ActiveRequest>>, llvm::DenseMapInfo<swift::ActiveRequest, void>>::empty() const /path/to/llvm-project/llvm/include/llvm/ADT/DenseSet.h:86:38
#50 0x000001fcbd36de83 swift::ExecuteSILPipelineRequest::OutputType swift::Evaluator::getResultUncached<swift::ExecuteSILPipelineRequest, swift::ExecuteSILPipelineRequest::OutputType swift::evaluateOrFatal<swift::ExecuteSILPipelineRequest>(swift::Evaluator&, swift::ExecuteSILPipelineRequest)::'lambda'()>(swift::ExecuteSILPipelineRequest const&, swift::ExecuteSILPipelineRequest::OutputType swift::evaluateOrFatal<swift::ExecuteSILPipelineRequest>(swift::Evaluator&, swift::ExecuteSILPipelineRequest)::'lambda'())::'lambda'()::operator()() const /path/to/swift/include/swift/AST/Evaluator.h:359:34
#51 0x000001fcbd36de83 swift::ExecuteSILPipelineRequest::OutputType swift::Evaluator::getResultUncached<swift::ExecuteSILPipelineRequest, swift::ExecuteSILPipelineRequest::OutputType swift::evaluateOrFatal<swift::ExecuteSILPipelineRequest>(swift::Evaluator&, swift::ExecuteSILPipelineRequest)::'lambda'()>(swift::ExecuteSILPipelineRequest const&, swift::ExecuteSILPipelineRequest::OutputType swift::evaluateOrFatal<swift::ExecuteSILPipelineRequest>(swift::Evaluator&, swift::ExecuteSILPipelineRequest)::'lambda'()) /path/to/swift/include/swift/AST/Evaluator.h:354:19
#52 0x000001fcbd35e5ff swift::executePassPipelinePlan(swift::SILModule*, swift::SILPassPipelinePlan const&, bool, swift::irgen::IRGenModule*) /path/to/swift/lib/SILOptimizer/PassManager/PassManager.cpp:404:1
#53 0x000001fcbd36f4fb std::_Vector_base<swift::SILPassPipeline, std::allocator<swift::SILPassPipeline>>::~_Vector_base() /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/stl_vector.h:375:54
#54 0x000001fcbd36f4fb std::vector<swift::SILPassPipeline, std::allocator<swift::SILPassPipeline>>::~vector() /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/stl_vector.h:795:7
#55 0x000001fcbd36f4fb swift::SILPassPipelinePlan::~SILPassPipelinePlan() /path/to/swift/include/swift/SILOptimizer/PassManager/PassPipeline.h:73:34
#56 0x000001fcbd36f4fb swift::runSILOptimizationPasses(swift::SILModule&) /path/to/swift/lib/SILOptimizer/PassManager/Passes.cpp:123:3
#57 0x000001fcbc7f79f1 performSILOptimizations(swift::CompilerInvocation&, swift::SILModule*) /path/to/swift/lib/Frontend/Frontend.cpp:1871:7
#58 0x000001fcbc7f79f1 swift::CompilerInstance::performSILProcessing(swift::SILModule*) /path/to/swift/lib/Frontend/Frontend.cpp:1900:3
#59 0x000001fcbc462ed6 performCompileStepsPostSILGen(swift::CompilerInstance&, std::unique_ptr<swift::SILModule, std::default_delete<swift::SILModule>>, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, swift::PrimarySpecificPaths const&, int&, swift::FrontendObserver*, llvm::ArrayRef<char const*>, llvm::ArrayRef<swift::PrimarySpecificPaths>) /path/to/swift/lib/FrontendTool/FrontendTool.cpp:2200:16
#60 0x000001fcbc461a9f swift::performCompileStepsPostSema(swift::CompilerInstance&, int&, swift::FrontendObserver*, llvm::ArrayRef<char const*>) /path/to/swift/lib/FrontendTool/FrontendTool.cpp:885:12
#61 0x000001fcbc47a369 withSemanticAnalysis(swift::CompilerInstance&, swift::FrontendObserver*, llvm::function_ref<bool (swift::CompilerInstance&)>, bool) /path/to/swift/lib/FrontendTool/FrontendTool.cpp:1328:25
#62 0x000001fcbc467deb performAction(swift::CompilerInstance&, int&, swift::FrontendObserver*, llvm::ArrayRef<char const*>) /path/to/swift/lib/FrontendTool/FrontendTool.cpp:0:12
#63 0x000001fcbc467deb performCompile(swift::CompilerInstance&, int&, swift::FrontendObserver*, llvm::ArrayRef<char const*>) /path/to/swift/lib/FrontendTool/FrontendTool.cpp:1854:7
#64 0x000001fcbc4643c2 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) /path/to/swift/lib/FrontendTool/FrontendTool.cpp:0:19
#65 0x000001fcbc162f94 run_driver(llvm::StringRef, llvm::ArrayRef<char const*>, llvm::ArrayRef<char const*>) /path/to/swift/lib/DriverTool/driver.cpp:0:0
#66 0x000001fcbc162f94 swift::mainEntry(int, char const**) /path/to/swift/lib/DriverTool/driver.cpp:530:10
#67 0x0000644c9362777e __libc_start_call_main /usr/src/debug/glibc/glibc/csu/../sysdeps/nptl/libc_start_call_main.h:83:3
#68 0x0000644c936278bc call_init /usr/src/debug/glibc/glibc/csu/../csu/libc-start.c:128:20
#69 0x0000644c936278bc __libc_start_main /usr/src/debug/glibc/glibc/csu/../csu/libc-start.c:347:5
#70 0x000001fcbc1620a5 _start (../build/Ninja-RelWithDebInfoAssert/swift-linux-x86_64/bin/swift-frontend+0x7710a5)
```

### Expected behavior

No crash

### Environment

Swift version 6.4-dev (LLVM 155bd39530126ad, Swift 2e033e9bb76b0d4)
Target: x86_64-unknown-linux-gnu
Build config: +assertions

### Additional information

_No response_

--- a/SwiftCompilerSources/Sources/Optimizer/InstructionSimplification/SimplifyDifferentiableFunction.swift
+++ b/SwiftCompilerSources/Sources/Optimizer/InstructionSimplification/SimplifyDifferentiableFunction.swift
@@ -103,22 +103,36 @@ extension DifferentiableFunctionInst : SILCombineSimplifiable {
         continue
       }
 
+      // If the extractee has non-trivial ownership, it is consumed by the differentiable_function instruction.
+      // We must copy it before the consumption point so the copy remains live afterward.
+      let effectiveExtractee: Value
+      let needsDestroy: Bool
+      if extractee.ownership != .none {
+        let copyBuilder = Builder(before: self, context)
+        effectiveExtractee = copyBuilder.createCopyValue(operand: extractee)
+        needsDestroy = true
+      } else {
+        effectiveExtractee = extractee
+        needsDestroy = false
+      }
+
       switch differentiableFunctionExtract.ownership {
       case .none:
-        if differentiableFunctionExtract.type != extractee.type {
+        if differentiableFunctionExtract.type != effectiveExtractee.type {
           let convertBuilder = Builder(before: differentiableFunctionExtract, context)
-          let newField = convertBuilder.createConvertFunction(originalFunction: extractee, resultType: differentiableFunctionExtract.type, withoutActuallyEscaping: false)
+          let newField = convertBuilder.createConvertFunction(originalFunction: effectiveExtractee, resultType: differentiableFunctionExtract.type, withoutActuallyEscaping: false)
           differentiableFunctionExtract.replace(with: newField, context)
         } else {
-          differentiableFunctionExtract.replace(with: extractee, context)
+          differentiableFunctionExtract.replace(with: effectiveExtractee, context)
         }
 
       case .guaranteed:
         let beginBuilder = Builder(before: beginBorrow, context)
-        let borrowedField = beginBuilder.createBeginBorrow(of: extractee,
+        let borrowedField = beginBuilder.createBeginBorrow(of: effectiveExtractee,
                                                            isLexical: beginBorrow.isLexical,
                                                            hasPointerEscape: beginBorrow.hasPointerEscape)
-        if differentiableFunctionExtract.type != extractee.type {
+
+        if differentiableFunctionExtract.type != effectiveExtractee.type {
           let convertBuilder = Builder(before: differentiableFunctionExtract, context)
           let newField = convertBuilder.createConvertFunction(originalFunction: borrowedField, resultType: differentiableFunctionExtract.type, withoutActuallyEscaping: false)
           differentiableFunctionExtract.replace(with: newField, context)
@@ -128,6 +142,9 @@ extension DifferentiableFunctionInst : SILCombineSimplifiable {
         for endBorrow in beginBorrow.endInstructions {
           let endBuilder = Builder(before: endBorrow, context)
           endBuilder.createEndBorrow(of: borrowedField)
+          if needsDestroy {
+            endBuilder.createDestroyValue(operand: effectiveExtractee)
+          }
         }
 
       case .owned, .unowned:

--- a/test/AutoDiff/SILOptimizer/sil_combine_differentiable_function_extract.sil
+++ b/test/AutoDiff/SILOptimizer/sil_combine_differentiable_function_extract.sil
@@ -34,6 +34,8 @@ sil @$s3src3fooyxx_xts18AdditiveArithmeticRz16_Differentiation14DifferentiableRz
 // reverse-mode derivative of foo<A>(_:_:)
 sil @$s3src3fooyxx_xts18AdditiveArithmeticRz16_Differentiation14DifferentiableRzlFsACRzAdERzlTJrSSpSr : $@convention(thin) <τ_0_0 where τ_0_0 : AdditiveArithmetic, τ_0_0 : Differentiable> (@in_guaranteed τ_0_0, @in_guaranteed τ_0_0) -> (@out τ_0_0, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @out τ_0_2) for <τ_0_0.TangentVector, τ_0_0.TangentVector, τ_0_0.TangentVector>)
 
+// CHECK-LABEL: {{^}}// reverse-mode derivative of bar(_:_:){{$}}
+
 // reverse-mode derivative of bar(_:_:)
 sil [ossa] @$s3src3baryS2f_SftFTJrSSpSr : $@convention(thin) (Float, Float) -> (Float, @owned @callee_guaranteed (Float) -> (Float, Float)) {
 [global: read,write,copy,destroy,allocate,deinit_barrier]
@@ -97,3 +99,136 @@ sil @$s3src3baryS2f_SftFTJpSSpSr : $@convention(thin) (Float, @owned @callee_gua
 // thunk for @escaping @callee_guaranteed (@in_guaranteed Float) -> (@out Float, @out Float)
 sil [transparent] [reabstraction_thunk] @$sS3fIegnrr_S3fIegydd_TR : $@convention(thin) (Float, @guaranteed @callee_guaranteed (@in_guaranteed Float) -> (@out Float, @out Float)) -> (Float, Float) 
 
+
+/// SIL below corresponds to the following Swift:
+/// func compute<T>(_ f: (T) -> T, _ x: T) -> T {
+///   return f(x)
+/// }
+///
+/// func checkDerivative<T: Differentiable>(
+///   _ f: @differentiable(reverse) (T) -> T,
+///   _ x: T) {
+///   let _ = compute(f, x)
+/// }
+///
+/// func caller() { checkDerivative({ $0 }, 0.0) }
+
+// closure #1 in caller()
+// Isolation: nonisolated
+sil [ossa] @$s3src6callerSdyFS2dcfU_ : $@convention(thin) (Double) -> Double
+
+// forward-mode derivative of closure #1 in caller()
+// Isolation: nonisolated
+sil [signature_optimized_thunk] [heuristic_always_inline] @$s3src6callerSdyFS2dcfU_TJfSpSr : $@convention(thin) (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)
+
+// reverse-mode derivative of closure #1 in caller()
+// Isolation: nonisolated
+sil @$s3src6callerSdyFS2dcfU_TJrSpSr : $@convention(thin) (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)
+
+// thunk for @callee_guaranteed (@unowned Double) -> (@unowned Double, @owned @escaping @callee_guaranteed (@unowned Double) -> (@unowned Double))
+sil [transparent] [serialized] [reabstraction_thunk] [ossa] @$sS4dIegyd_Igydo_S2dxq_Ri_zRi0_zRi__Ri0__r0_lyS2dIsegnr_Iegnro_TR : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)) -> (@out Double, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>)
+
+// thunk for @callee_guaranteed (@unowned Double) -> (@unowned Double)
+sil [transparent] [serialized] [reabstraction_thunk] [ossa] @$sS2dIgyd_S2dIegnr_TR : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> Double) -> @out Double
+
+sil_scope 1 { parent @$s3src6callerSdyF : $@convention(thin) () -> Double }
+sil_scope 2 { parent 1 }
+sil_scope 3 { parent @$s3src15checkDerivativeyyxxYjrXE_xt16_Differentiation14DifferentiableRzlFSd_Tg5 : $@convention(thin) (@guaranteed @differentiable(reverse) @noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>, Double) -> () inlined_at 2 }
+sil_scope 4 { parent 3 inlined_at 2 }
+sil_scope 5 { parent 4 inlined_at 2 }
+sil_scope 6 { parent @$s3src7computeyxxxXE_xtlFSd_Tg5 : $@convention(thin) (@guaranteed @noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>, Double) -> Double inlined_at 5 }
+
+// CHECK-LABEL: {{^}}// caller(){{$}}
+
+// caller()
+// Isolation: unspecified
+sil [ossa] @$s3src6callerSdyF : $@convention(thin) () -> Double {
+[global: read,write,copy,destroy,allocate,deinit_barrier]
+bb0:
+  %0 = alloc_stack $Double // users: %37, %43, %42
+  // function_ref closure #1 in caller()
+  %1 = function_ref @$s3src6callerSdyFS2dcfU_ : $@convention(thin) (Double) -> Double // user: %4
+  // function_ref forward-mode derivative of closure #1 in caller()
+  %2 = function_ref @$s3src6callerSdyFS2dcfU_TJfSpSr : $@convention(thin) (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double) // user: %5
+  // function_ref reverse-mode derivative of closure #1 in caller()
+  %3 = function_ref @$s3src6callerSdyFS2dcfU_TJrSpSr : $@convention(thin) (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double) // user: %6
+  %4 = thin_to_thick_function %1 to $@noescape @callee_guaranteed (Double) -> Double // users: %9, %7; ownership: none
+  %5 = thin_to_thick_function %2 to $@noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double) // users: %15, %7; ownership: none
+  %6 = thin_to_thick_function %3 to $@noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double) // users: %21, %7; ownership: none
+  %7 = differentiable_function [parameters 0] [results 0] %4 with_derivative {%5, %6} // user: %40
+
+  // function_ref thunk for @callee_guaranteed (@unowned Double) -> (@unowned Double)
+  %8 = function_ref @$sS2dIgyd_S2dIegnr_TR : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> Double) -> @out Double // user: %9
+  // CHECK: %[[#A0:]] = function_ref @$sS2dIgyd_S2dIegnr_TR : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> Double) -> @out Double // users: %[[#]], %[[#]]{{$}}
+
+  %9 = partial_apply [callee_guaranteed] %8(%4) : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> Double) -> @out Double // user: %10
+  // CHECK: %[[#A1:]] = partial_apply [callee_guaranteed] %[[#A0]](%[[#A2:]]) : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> Double) -> @out Double // user: %[[#A3:]]
+
+  %10 = convert_function %9 to $@callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double> // users: %12, %11
+  // CHECK: %[[#A3:]] = convert_function %[[#A1]] to $@callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double> // users: %[[#]], %[[#]]{{$}}
+
+  %11 = copy_value %10 // users: %44, %13
+  // CHECK: %[[#A4:]] = copy_value %[[#A3]] // users: %[[#]], %[[#]]{{$}}
+
+  destroy_value %10 // id: %12
+  // CHECK: destroy_value %[[#A3]] // id: %11
+
+  %13 = convert_escape_to_noescape %11 to $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double> // user: %26
+  // CHECK: %[[#A5:]] = convert_escape_to_noescape %[[#A4]] to $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double> // users: %[[#]], %[[#]], %[[#]]{{$}}
+
+  // function_ref thunk for @callee_guaranteed (@unowned Double) -> (@unowned Double, @owned @escaping @callee_guaranteed (@unowned Double) -> (@unowned Double))
+  %14 = function_ref @$sS4dIegyd_Igydo_S2dxq_Ri_zRi0_zRi__Ri0__r0_lyS2dIsegnr_Iegnro_TR : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)) -> (@out Double, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>) // user: %15
+  %15 = partial_apply [callee_guaranteed] %14(%5) : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)) -> (@out Double, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>) // user: %16
+  %16 = convert_function %15 to $@callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double> // users: %18, %17
+  %17 = copy_value %16 // users: %45, %19
+  destroy_value %16 // id: %18
+  %19 = convert_escape_to_noescape %17 to $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double> // user: %26
+  // function_ref thunk for @callee_guaranteed (@unowned Double) -> (@unowned Double, @owned @escaping @callee_guaranteed (@unowned Double) -> (@unowned Double))
+  %20 = function_ref @$sS4dIegyd_Igydo_S2dxq_Ri_zRi0_zRi__Ri0__r0_lyS2dIsegnr_Iegnro_TR : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)) -> (@out Double, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>) // user: %21
+  %21 = partial_apply [callee_guaranteed] %20(%6) : $@convention(thin) (@in_guaranteed Double, @guaranteed @noescape @callee_guaranteed (Double) -> (Double, @owned @callee_guaranteed (Double) -> Double)) -> (@out Double, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>) // user: %22
+  %22 = convert_function %21 to $@callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double> // users: %24, %23
+  %23 = copy_value %22 // users: %46, %25
+  destroy_value %22 // id: %24
+  %25 = convert_escape_to_noescape %23 to $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1, τ_0_2, τ_0_3> (@in_guaranteed τ_0_0) -> (@out τ_0_1, @owned @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <τ_0_2, τ_0_3>) for <Double, Double, Double, Double> // user: %26
+
+  // CHECK: %[[#A6:]] = copy_value %[[#A5]]
+  %26 = differentiable_function [parameters 0] [results 0] %13 with_derivative {%19, %25} // users: %29, %41
+  // CHECK: %[[#A7:]] = differentiable_function [parameters 0] [results 0] %[[#A6]] with_derivative {%[[#]], %[[#]]} // users: %[[#]], %[[#]]{{$}}
+
+  %27 = float_literal $Builtin.FPIEEE64, 0x0 // user: %28
+  %28 = struct $Double (%27) // users: %30, %34
+  %29 = begin_borrow [lexical] %26 // users: %32, %31, %39
+  debug_value %28, let, name "x", argno 2, scope 3 // id: %30
+  debug_value %29, let, name "f", argno 1, scope 3 // id: %31
+  %32 = differentiable_function_extract [original] %29 // users: %37, %35
+
+  %33 = alloc_stack $Double // users: %38, %37, %36, %34
+  store %28 to [trivial] %33 // id: %34
+
+  debug_value %32, let, name "f", argno 1, scope 5 // id: %35
+  // CHECK: debug_value %[[#A5]]
+  // CHECK: debug_value %[[#A7]]
+
+  debug_value %33, let, name "x", argno 2, expr op_deref, scope 5 // id: %36
+
+  // CHECK: %[[#A8:]] = alloc_stack $Double
+  %37 = apply %32(%0, %33) : $@noescape @callee_guaranteed @substituted <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0) -> @out τ_0_1 for <Double, Double>
+  // CHECK: %[[#]] = apply %[[#A0]](%0, %[[#A8]], %[[#A2]])
+
+  dealloc_stack %33 // id: %38
+  end_borrow %29 // id: %39
+  destroy_value %7 // id: %40
+
+  destroy_value %26 // id: %41
+  // CHECK: destroy_value %[[#A7]]
+
+  %42 = load [trivial] %0 // user: %47
+  dealloc_stack %0 // id: %43
+
+  destroy_value %11 // id: %44
+  // CHECK: destroy_value %[[#A4]]
+
+  destroy_value %17 // id: %45
+  destroy_value %23 // id: %46
+  return %42 // id: %47
+} // end sil function '$s3src6callerSdyF'

